packages/waku/src/router/client.ts

```diff
@@ -342,6 +346,58 @@ export class ErrorBoundary extends Component<
   }
 }
 
+const NotFound = () => {
+  // FIXME totally unsure if this is a desired behavior
+  useEffect(() => {
+    window.location.replace('/404');
+  });
+  return createElement(
+    'html',
+    null,
+    createElement('body', null, createElement('h1', null, 'Not Found')),
+  );
+};
+
+const Redirect = ({ loc }: { loc: string }) => {
+  // FIXME unsure if this is a desired behavior
+  useEffect(() => {
+    window.location.replace(loc);
+  });
+  return createElement(
+    'html',
+    null,
+    createElement('body', null, createElement('a', { href: loc }, 'Redirect')),
+  );
+};
+
+class CustomErrorHandler extends Component<
```

Comment:
Can we add a way to pass in components to use instead of the defaults? I want to be able to define my own NotFound component, for example. Maybe we can pass it through as a `<Router />` prop or another method?

https://github.com/vercel/next.js/blob/v15.1.0/packages/next/src/client/components/http-access-fallback/error-boundary.tsx#L169-L177

Reply:
I think you would like to use an error that isn't captured by CustomErrorHandler. Let's revisit it as our design doesn't seem very solid.

Reply:
This comment was about being able to create different element output than `<html><body><h1>Not Found</h1></body></html>` component when there is a stream. If it can render a [404 page component if available](https://github.com/dai-shi/waku/pull/1261/files#r1967194656), that works.